### DISCUSSION TITLE
Sonatype requires tokens now

### DISCUSSION THREAD

**Discussion by hazendaz** [spotbugs/spotbugs#3018] [General]
See https://github.com/spotbugs/spotbugs/commit/760571c6808c46203213ba0552d8cac050f20cd3

Sonatype no longer allows user name / password.  Not sure who had that setup originally but its under me now.  Not sure if the release process is similar on the other repos or not as I did not check. I do the maven plugin manually so nothing to do there.  The main spotbugs here is fixed now so it releases again after adjusting for that as I already knew about this for a few days as sonatype emailed me.  Anyway, if we need to change elsewhere, I can just do as I did here for sake of making it easy.  Do let me know if there are other spots I should look.

I do think we need to also relax the gradle plugin as it releases far too quickly for nonsense and at lease one user has raised issue.  I think we should release it when we need to more like we are doing with everything else so I think that means removing some of its foo automation as a user can construct a special commit there to even cause a release which in my opinion is a security issue.  But also its not usual I don't think to do such a thing so its a bit confusing when it does release.  I also hate its spam email of every single little thing that happened being emailed out.  I delete those, I'm sure everyone else does too.  Just mentioning it here for lack of better place to do so since its going to release in a day due to spotbugs release tonight.

**Comment by JuditKnoll** [comment]
Originally it was under @KengoTODA's name, at least the earlier username is mentioned on his GitHub page.
Thank you for updating it, @hazendaz 😄 
From the [sonatype guide](https://central.sonatype.org/publish/generate-token/), aside from updating the secret from password to token, nothing else needs to be changed. From the successful release, it looks like it works like charm 😄 

As far as I can see, Sonatype is used for 4 projects in the organization:
- spotbugs/spotbugs, which you just updated,
- [spotbugs-maven-plugin](https://github.com/spotbugs/spotbugs-maven-plugin)
   - [the workflow](https://github.com/spotbugs/spotbugs-maven-plugin/blob/master/.github/workflows/sonatype.yaml#L23)
   - From the blame it looks like you were the one to add these env vars. Can you please update the secrets?
   - 6/24/2024 - the workflow is already tokenized there, reconfirmed.
- [sonar-findbugs](https://github.com/spotbugs/sonar-findbugs)
   - [release workflow](https://github.com/spotbugs/sonar-findbugs/blob/master/.github/workflows/build.yml#L104)
   - It's under @KengoTODA's name. As far as I know, @gtoison is the most active in that repo, so he should have a say in this, but 
   - 6/24/2204 - per https://github.com/spotbugs/sonar-findbugs/issues/1027, done but uses a variation closer to what it is since original credential was listed as jira.
IMO it's best if all the repos in the org are under the same sonatype user. @gtoison What do you think?
- [spotbugs-archetype](https://github.com/spotbugs/spotbugs-archetype)
   - [the workflow file](https://github.com/spotbugs/spotbugs-archetype/blob/master/.github/workflows/build.yml#L42)
   - It's under @KengoTODA's name.

In the other repositories, I could not find anything for "sonatype" or "ossrh", so I think those don't use it.

**Reply by gtoison** [reply]
I'm happy with using the same sonatype user, I've created and issue here for the sonar plugin: https://github.com/spotbugs/sonar-findbugs/issues/1027

**Comment by JuditKnoll** [comment]
As far as I can see, this discussion got resolved, so I'm closing it.
